inc/item_devicesimcard.class.php

```diff
@@ -0,0 +1,85 @@
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+* @brief
+* @since version 9.2
+*/
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+/**
+ * Relation between item and devices
+ **/
+class Item_DeviceSimcard extends Item_Devices {
+
+   static public $itemtype_2 = 'DeviceSimcard';
+   static public $items_id_2 = 'devicesimcards_id';
+
+   static protected $notable = false;
+
+   /**
+    * @since version 0.85
+    **/
+   static function getSpecificities($specif='') {
+      return [
+             'serial'         => parent::getSpecificities('serial'),
+             'otherserial'    => parent::getSpecificities('otherserial'),
+             'locations_id'   => parent::getSpecificities('locations_id'),
+             'states_id'      => parent::getSpecificities('states_id'),
+             'pin'            => array('long name'  => __('PIN code'),
+                                       'short name' => __('PIN code'),
+                                       'size'       => 20,
+                                       'id'         => 15),
+            'pin2'            => array('long name'  => __('PIN2 code'),
+                                       'short name' => __('PIN2 code'),
+                                       'size'       => 20,
+                                       'id'         => 16),
+            'puk'             => array('long name'  => __('PUK code'),
+                                       'short name' => __('PUK code'),
+                                       'size'       => 20,
+                                       'id'         => 17),
+            'puk2'            => array('long name'  => __('PUK2 code'),
+                                       'short name' => __('PUK2 code'),
+                                       'size'       => 20,
+                                       'id'         => 18),
+            'lines_id'        => array('long name'  => __('Line'),
+                                       'short name' => __('Line'),
+                                       'size'       => 20,
+                                       'id'         => 19,
+                                       'datatype'   => 'dropdown'),
+      ];
```

Comment:
Encrypting these fields has a limited interest because we need a reversible encryption. Users having abilities to access the database for example have very likely enough skills to easily find from the soruce how to decrypt these data.